homeassistant/components/anthropic/conversation.py

```diff
@@ -266,34 +366,58 @@ async def _async_handle_message(
         system = chat_log.content[0]
         if not isinstance(system, conversation.SystemContent):
             raise TypeError("First message must be a system message")
-        messages = [_convert_content(content) for content in chat_log.content[1:]]
+        messages = _convert_content(chat_log.content[1:])
 
         client = self.entry.runtime_data
 
+        thinking_budget = options.get(CONF_THINKING_BUDGET, RECOMMENDED_THINKING_BUDGET)
+        model = options.get(CONF_CHAT_MODEL, RECOMMENDED_CHAT_MODEL)
+
         # To prevent infinite loops, we limit the number of iterations
         for _iteration in range(MAX_TOOL_ITERATIONS):
+            if model in THINKING_MODELS and thinking_budget >= MIN_THINKING_BUDGET:
```

Comment:
Why check the min budget, isn't that checked in the config flow already? Wouldn't the system fail if you try to interact with a thinking model without a budget?

Reply:
It would fail:
![image](https://github.com/user-attachments/assets/037a437e-e01b-4b62-aed6-f8bb0965c009)
We check it to send the enabled/disabled flag. If it is less than 1024, we consider it disabled.
No, we don't check that in config flow anymore.